File: 3174 clear-digits/solution.cpp
class Solution {
public:
    string clearDigits(string s) {
        vector<char> st(s.size());
        int top = -1;
        for(char c:s) {
            if(top!=-1 && isdigit(c)) top--;
            else st[++top] = c;
        }
        return std::string(st.begin(), st.begin() + top + 1);

    }
};

File: 3174 clear-digits/solution.cs
public class Solution {
    public string ClearDigits(string s) {
        char[] st = new char[s.Length];
        int top = -1;
        foreach(char c in s) {
            if(top!=-1 && Char.IsDigit(c)) top--;
            else st[++top] = c;
        }
        return new string(st,0,top+1);
    }
}
